frontend/src/stores/ui.ts

```diff
@@ -17,6 +17,15 @@ import { needsTranslationBanner } from "~/utils/translation-banner"
 
 const desktopBreakpoints: RealBreakpoint[] = ["2xl", "xl", "lg"]
 
+export type SnackbarState = "not_shown" | "visible" | "dismissed"
+export type ColorMode = "dark" | "light" | "system"
+
+export function isColorMode(value: undefined | string): value is ColorMode {
+  return (
+    typeof value === "string" && ["light", "dark", "system"].includes(value)
+  )
+}
+
```

Comment:
I moved these types in here; it makes more sense to me that the UI Store is the "source of truth" for UI state, and cookies are an implementation of a sub-set of that state.